compiler/expressions.cpp

```diff
@@ -1896,6 +1896,18 @@ SC3ExpressionParser::parse_view_as(value* lval)
     return lvalue;
 }
 
+bool
+is_valid_index_tag(int tag)
+{
+    Type* idx_type = gTypes.find(tag);
+    if (idx_type->isStruct() || idx_type->isFunction() || idx_type->isEnumStruct() ||
+        idx_type->isMethodmap() || tag == pc_tag_void || tag == sc_rationaltag)

```

Comment:
`pc_tag_string`?  I dunno if there's a better way than listing these all out.

Reply:
I couldn't decide if pc_tag_string should be included, indexing by char sounds like a legit thing.